src/Plugins/SimplnxCore/src/SimplnxCore/Filters/CalculateFeatureSizesFilter.cpp

```diff
@@ -291,6 +296,101 @@ Result<> CalculateFeatureSizesFilter::executeImpl(DataStructure& dataStructure,
     }
   }
 
+  // Calculate the Feret Diameters (Caliper diameters)
+  {
+    auto featureAttrMat = dataStructure.getDataRefAs<AttributeMatrix>(featureAttributeMatrixPath);
+    auto numFeatures = featureAttrMat.getNumTuples();
+
+    std::vector<uint32> corners(numFeatures * 6);
+    // Create corners array, which stores pixel coordinates for the top-left and bottom-right coordinates of each feature object
+    for(usize i = 0; i < numFeatures; i++)
+    {
+      corners[i * 6 + 0] = std::numeric_limits<uint32>::max();
+      corners[i * 6 + 1] = std::numeric_limits<uint32>::max();
+      corners[i * 6 + 2] = std::numeric_limits<uint32>::max();
+      corners[i * 6 + 3] = std::numeric_limits<uint32>::min();
+      corners[i * 6 + 4] = std::numeric_limits<uint32>::min();
+      corners[i * 6 + 5] = std::numeric_limits<uint32>::min();
+    }
+    auto imageDims = geom->getDimensions();
+
+    const usize xDim = imageDims[0];
+    const usize yDim = imageDims[1];
+    const usize zDim = imageDims[2];
+
+    usize index = 0;
+    // Store the coordinates in the corners array
+    for(uint32 z = 0; z < zDim; z++)
+    {
+      if(shouldCancel)
+      {
+        return {};
+      }
+
+      for(uint32 y = 0; y < yDim; y++)
+      {
+        for(uint32 x = 0; x < xDim; x++)
+        {
+          index = (imageDims[1] * imageDims[0] * z) + (imageDims[0] * y) + x; // Index into featureIds array
```

Comment:
Looks like this is only used inside the loop so the declaration can just become this line.